sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -222,13 +222,13 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession with SQLHelper
       // Create a test case to ignore this case.
       ignore(testCase.name) { /* Do nothing */ }
     } else testCase match {
-      case udfTestCase: UDFTest
+      case udfTestCase: SQLQueryTestSuite#UDFTest
```

Comment:
```
[ERROR] [Error] /spark/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala:225: The outer reference in this type test cannot be checked at run time.
[ERROR] [Error] /spark/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala:357: The outer reference in this type test cannot be checked at run time.
[ERROR] [Error] /spark/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala:363: The outer reference in this type test cannot be checked at run time.
[ERROR] [Error] /spark/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala:371: The outer reference in this type test cannot be checked at run time.
[ERROR] [Error] /spark/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala:414: The outer reference in this type test cannot be checked at run time.
```

Reply:
@som-snytt It seems that this compilation fix is related [https://github.com/scala/scala/pull/9504](https://github.com/scala/scala/pull/9504).

So I have another question to ask you for help, `UDFTest`  is an internal `trait` defined in `SQLQueryTestSuite`,  must full path be used for this case in future Scala versions?

Reply:
I don't have a precise answer, but I would expect ongoing improvements and also some noise. You can use `-Wconf` to suppress (or escalate) the message. I will try to take another look tomorrow; I haven't followed the pattern matching changes yet.

Reply:
thanks, I tried to suppress this behavior in the past, but it seems that this is a compilation error now, I'll try it again